test/t8n/t8n.cpp

```diff
@@ -19,6 +19,64 @@ using namespace evmone;
 using namespace evmone::test;
 using namespace std::literals;
 
+namespace
+{
+using namespace intx;
+
+int64_t get_bomb_delay(evmc_revision rev)
+{
+    switch (rev)
+    {
+    case EVMC_BYZANTIUM:
+        return 3000000;
+    case EVMC_CONSTANTINOPLE:
+    case EVMC_PETERSBURG:
+    case EVMC_ISTANBUL:
+        return 5000000;
+    case EVMC_BERLIN:
+        return 9000000;
+    case EVMC_LONDON:
+        return 9700000;
+    default:
+        throw std::runtime_error("get_bomb_delay: Wrong rev");
+    }
+}
+
+int64_t calc_difficulty(const int64_t& parent_difficulty, const hash256& parent_uncle_hash,
+    const int64_t& parent_timestamp, const int64_t& current_timestamp, const int64_t& block_num,
+    evmc_revision rev)
+{
+    if (rev >= EVMC_PARIS)
+        return 0;
+
+    // TODO: Implement for older revisions
+    if (rev < EVMC_BYZANTIUM)
+        return 0x020000;
+
+    static constexpr auto min_difficulty = int64_t{1} << 17;
+
+    const auto kappa = get_bomb_delay(rev);
+
+    const auto H_i_prime = kappa >= block_num ? 0 : block_num - kappa;
+
+    const auto p = (H_i_prime / 100000) - 2;
+    assert(p < 63);
+
+    const auto epsilon = p < 0 ? 0 : int64_t{1} << p;
+
+    static const hash256 empty_uncle_hash =
+        0x1dcc4de8dec75d7aab85b567b6ccd41ad312451b948a7413f0a142fd40d49347_bytes32;
```

Comment:
I think this is a hash of empty RLP list. See `EmptyLogsHash`.

Reply:
Created `EmptyListHash` in `state.hpp`. Later it can be used instead `EmptyLogsHash`

Reply:
We should get #699 first.